Polyfills/Window/Source/Window.cpp

```diff
@@ -72,17 +72,50 @@ namespace Babylon::Polyfills::Internal
     Window::Window(const Napi::CallbackInfo& info)
         : Napi::ObjectWrap<Window>{info}
         , m_runtime{JsRuntime::GetFromJavaScript(info.Env())}
+        , m_timeoutDispatcher{m_runtime}
     {
     }
 
-    void Window::SetTimeout(const Napi::CallbackInfo& info)
+    Napi::Value Window::SetTimeout(const Napi::CallbackInfo& info)
     {
-        auto function = Napi::Persistent(info[0].As<Napi::Function>());
-        auto milliseconds = std::chrono::milliseconds{info[1].As<Napi::Number>().Int32Value()};
-
         auto& window = *static_cast<Window*>(info.Data());
 
-        window.RecursiveWaitOrCall(std::make_shared<Napi::FunctionReference>(std::move(function)), std::chrono::system_clock::now() + milliseconds);
+        const auto arg0 = info[0];
+        if (!arg0.IsFunction())
+        {
+            return Napi::Value::From(info.Env(), window.m_timeoutDispatcher->NextTimeoutId());
```

Comment:
It's highly unlikely, but this doesn't guarantee that a subsequent setTimeout doesn't use the same id if the id wraps around. Are you trying to return a positive number?

Reply:
We could return `undefined` instead, so that a call to `clearTimeout` using the value returned here does not do anything.

Reply:
You could also reserve a number (say max value) for this specific case. What do browsers do?

Reply:
Chrome returns the next id.

Reply:
Maybe it also adds an entry into the map so that it can collide with other ones?

Reply:
Probably. It's hard to tell for certain looking at the Chromium source code.

Reply:
Change in commit https://github.com/BabylonJS/BabylonNative/pull/1146/commits/6e605ccbe7648530237002cde4c03f829855264e  to use a function that does nothing.